src/CoreWCF.UnixDomainSocket/src/CoreWCF/Channels/UnixDomainSocketListenOptions.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Connections;
+using System;
+using System.Collections.Generic;
+using System.Security.Principal;
+using System.Threading.Tasks;
+
+namespace CoreWCF.Channels
+{
+    public class UnixDomainSocketListenOptions : IConnectionBuilder
+    {
+        // Some properties which were originally set via ConnectionOrientedTransportBindingElement (eg ConnectionBufferSize) make more sense to
+        // add to this class. If there are multiple endpoints using named pipes, on .NET Framework the first endpoint would establish the settings
+        // and subsequent ones would need validate that the settings matched. By moving the settings needed to be considered for a shared listener
+        // to NamedPipeListenOptions, we remove the need to reconcile multiple bindings and just set it on the classes which are responsible
+        // for that part of the IO.
+        private readonly List<Func<ConnectionDelegate, ConnectionDelegate>> _middleware = new List<Func<ConnectionDelegate, ConnectionDelegate>>();
+        private List<SecurityIdentifier> _allowedUsers;
+
+        internal UnixDomainSocketListenOptions(Uri baseAddress)
+        {
+            UnixDomainSocketUri.Validate(baseAddress);
+            BaseAddress = baseAddress;
+            FilePath = baseAddress.AbsolutePath;
+        }
+
+        public List<SecurityIdentifier> AllowedUsers
+        {
```

Comment:
These issues shouldn't block for preview, just need to document there will be changes here, but we do need to do something with this. We also should have a FileMode property so that it can be an error if the file already exists, eg with CreateNew, or removing an existing file and creating a new one if Create is specified. We should have posix permission properties too where you can specify the unix group owner and the oge permission that you would chmod the file to.

Reply:
https://github.com/CoreWCF/CoreWCF/issues/1194 <= closing as moving to separate issue